packages/blade/src/components/Button/BaseButton/StyledBaseButton.web.tsx

```diff
@@ -1,42 +1,37 @@
 import styled from 'styled-components';
+import getIn from '../../../utils/getIn';
+import makeMotionTime from '../../../utils/makeMotionTime';
 import getBaseButtonStyles from './getBaseButtonStyles';
 import type { StyledBaseButtonProps } from './StyledBaseButton.d';
 
 const StyledBaseButton = styled.button(
-  ({
-    activeBorderColor,
-    activeBackgroundColor,
-    defaultBorderColor,
-    minHeight,
-    spacing,
-    defaultBackgroundColor,
-    disabled,
-    focusBorderColor,
-    focusBackgroundColor,
-    focusRingColor,
-    hoverBorderColor,
-    hoverBackgroundColor,
-    isFullWidth,
-    borderWidth,
-    borderRadius,
-  }: Omit<StyledBaseButtonProps, 'children' | 'onClick'>) =>
-    getBaseButtonStyles({
-      activeBorderColor,
-      activeBackgroundColor,
-      defaultBorderColor,
-      minHeight,
-      spacing,
-      defaultBackgroundColor,
-      disabled,
-      focusBorderColor,
-      focusBackgroundColor,
-      focusRingColor,
-      hoverBorderColor,
-      hoverBackgroundColor,
-      isFullWidth,
-      borderWidth,
-      borderRadius,
+  ({ ...props }: Omit<StyledBaseButtonProps, 'children' | 'onClick'>) => ({
+    ...getBaseButtonStyles({
+      ...props,
     }),
+    transitionProperty: 'background-color, border-color, box-shadow',
+    transitionTimingFunction: getIn(props.theme.motion, props.motionEasing),
+    transitionDuration: makeMotionTime(getIn(props.theme.motion, props.motionDuration)),
```

Comment:
i think this function mostly is used where you have something like string with dot notation. over here can't we directly do like below
```suggestion
    transitionDuration: makeMotionTime(props.theme.motion[props.motionDuration]),
```